src/to_ir.rs

```diff
@@ -15,6 +15,58 @@ use std::rc::Rc;
 // FilterVisitor:  visits tree and fills out structural and property filters
 /***********************************/
 
+pub struct PropertyVisitor {
+    property: Option<Property>,
+}
+
+impl Default for PropertyVisitor {
+    fn default() -> PropertyVisitor {
+        PropertyVisitor { property: None }
+    }
+}
+
+impl<'i> ParseTreeVisitor<'i, CypherParserContextType> for PropertyVisitor {
+    fn visit_terminal(&mut self, _node: &TerminalNode<'i, CypherParserContextType>) {}
+}
+
+impl<'i> CypherVisitor<'i> for PropertyVisitor {
+    fn visit_oC_PropertyOrLabelsExpression(
```

Comment:
Not super happy that we have to use a visitor for this because we have to access the result indirectly, but with this you can get a property project directly from an expression.